src/resolvers/solid.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  countSlash,
+  isCatchAllRoute,
+  isDynamicRoute,
+} from '../utils'
+import { generateClientCode } from '../stringify'
+
+import type { Optional, ResolvedOptions } from '../types'
+import type { PageContext } from '../context'
+
+export interface Route {
+  rawRoute: string
+  path: string
+  children?: Route[]
+  component?: string
+  element?: string
+}
+
+export type PrepareRoute = Omit<Optional<Route, 'rawRoute' | 'path'>, 'children'> & {
+  children?: PrepareRoute[]
+}
```

Comment:
I think is better to rename this two types like this, also in other resolves.
```suggestion
export interface SolidRouteBase {
  rawRoute: string
  path: string
  children?: Route[]
  component?: string
  element?: string
}

export type SolidRoute = Omit<Optional<SolidRouteBase, 'rawRoute' | 'path'>, 'children'> & {
  children?: PrepareRoute[]
}
```